swap/src/bob/swap.rs

```diff
@@ -357,13 +358,18 @@ where
                 .await
             }
             BobState::T1Expired(state4) => {
-                if state4
-                    .check_for_tx_cancel(bitcoin_wallet.as_ref())
-                    .await
-                    .is_err()
-                {
-                    state4.submit_tx_cancel(bitcoin_wallet.as_ref()).await?;
-                }
+                let result = state4.submit_tx_cancel(bitcoin_wallet.as_ref()).await;
```

Comment:
This should be done closer to the RPC call and we should avoid duplication this error handling code. I am preparing a patch to show how it can be done.

Reply:
https://github.com/comit-network/xmr-btc-swap/pull/103